src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/notifications/CaseNotesNotificationsService.kt

```diff
@@ -40,15 +40,7 @@ class CaseNotesNotificationsService(
 
   private fun emailResponsibleProbationPractitioner(referral: Referral, sender: AuthUser, caseNoteId: UUID) {
     val responsibleOfficer = referralService.getResponsibleProbationPractitioner(referral)
-    val senderIsResponsibleOfficer =
-      userTypeChecker.isProbationPractitionerUser(sender) &&
-        (
-          (responsibleOfficer.authUser?.let { it == sender } ?: false) ||
-            (
-              responsibleOfficer.deliusStaffId?.let { it == communityAPIOffenderService.getStaffIdentifier(sender) }
-                ?: false // if the RO doesn't have a staff ID we cannot determine if they are the sender, so assume not
-              )
-          )
+    val senderIsResponsibleOfficer = referralService.isUserTheResponsibleOfficer(responsibleOfficer, sender)
```

Comment:
I like this refactor, it highlights we are effectively asking how the `getResponsibleProbationPractitioner` algorithm decided

```kt
val ro = referralService.getResponsibleProbationPractitioner(...)
referralService.isUserTheResponsibleOfficer(ro, ...)
```

Since it's working backwards, what do you think about exposing this information from the returned object?

Maybe so that we can check `responsibleOfficer.role == RESPONSIBLE_OFFICER` or `responsibleOfficer.role == SENDER`, or similar? We can encode these decisions in the `getResponsibleProbationPractitioner` call.

This would also remove the need for all the new mocking below

Reply:
It is a very good point so that we can encapsulate the role in ResponsibleProbationPractitioner. I will make those changes.

Reply:
@sldblog , while refactoring this method, what is realised, if we want to find out something like this `responsibleOfficer.role == RESPONSIBLE_OFFICER or responsibleOfficer.role == SENDER`, we need to pass the other `user(sender)` as a parameter to the function. Then I feel the method is doing more than one thing like

- get the ProbabationPractioner
- Check whether the user passed is same as the probabationpractioner
- set the role.
I feel it's better off to leave it like that even if we want to do extra mocking.

Reply:
yes, agree. my mistake for not realising that we check if the delius entity is same as the sender